File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/visitor/solution/get_number_of_stops_journey_visitor.cpp
// get_number_of_stops_journey_visitor.cpp

#include "get_number_of_stops_journey_visitor.h"
class single_journey;

get_number_of_stops_journey_visitor::get_number_of_stops_journey_visitor()
: my_number_of_stops(0)
{
}

void get_number_of_stops_journey_visitor::visit(single_journey & a_single_journey)
{
	++my_number_of_stops;
}

int get_number_of_stops_journey_visitor::get_number_of_stops() const
{
	return my_number_of_stops;
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/visitor/solution/GetNumberOfStopsJourneyVisitor.java
// GetNumberOfStopsJourneyVisitor.java

public class GetNumberOfStopsJourneyVisitor extends JourneyVisitor
{
	public GetNumberOfStopsJourneyVisitor()
	{
		myNumberOfStops = 0;
	}

	public void visit(SingleJourney aSingleJourney)
	{
		++myNumberOfStops;
	}

	public int getNumberOfStops()
	{
		return myNumberOfStops;
	}

	private int myNumberOfStops;
}
